src/extension.ts

```diff
@@ -134,22 +134,14 @@ export async function activate(ctx: vscode.ExtensionContext) {
 
   const compileCommandsFile = `${buildDir}/compile_commands.json`;
   whenFileExists(ctx, compileCommandsFile, async () => {
-    try {
-      const conf = vscode.workspace.getConfiguration("C_Cpp");
-      conf.update("default.compileCommands", compileCommandsFile, vscode.ConfigurationTarget.Workspace);
-    } catch {
-      // Ignore, C/C++ extension might not be installed
-    }
+    const conf = vscode.workspace.getConfiguration("C_Cpp");
```

Comment:
What happens if this setting doesn't exist?

Reply:
Nothing happens. I have checked this and the VS Code configurations API seems to check if the setting we want to change is registered to be handled by some extension. The value is only written if such a handler exists.

No exceptions are thrown if an unsupported setting is changed and the `Thenable` return value is ignored.

The `Thenable` return value provides callbacks for handling both success and failure results, if any action is required.